### ISSUE TITLE
Crash using Mabbox iOS SDK(Annotation view)

### ISSUE THREAD

**Issue by macro-shen** [mapbox/mapbox-gl-native#9065]
a issue from a iOS developer:

**Platform:** iOS
**Mapbox SDK version:** 3.5.0

### Steps to trigger behavior

find a crash in Mabbox, and it often happens. The crash log is like this:
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Annotation view's annotation property should not be nil.'

Add a gesture to the map, but don't know the problem.
And there is an online log for this crash: [http://crashes.to/s/468bb46a57c](http://crashes.to/s/468bb46a57c)

For this bug, I have a problem, I code like this, the problem is: if the annotation is nil, there cannot be a Annotation view, so I don't know how could this bug happen.

```
- (MGLAnnotationView *) mapView:(MGLMapView *)mapView viewForAnnotation:(id<MGLAnnotation>)annotation {
    if( ![annotation isKindOfClass:[CDRunGroundAnnotationEntity class]] ) {
        return nil;
    }
}
```

**Comment by boundsj** [comment]
Hi @macro-shen this is most likely a duplicate of the issue described in https://github.com/mapbox/mapbox-gl-native/issues/8754 and https://github.com/mapbox/mapbox-gl-native/pull/8755. That issue was fixed in https://github.com/mapbox/mapbox-gl-native/pull/9025 and the fix will be in the our iOS map SDK v3.6.0 beta 1 release this week.

I'm going to close this as fixed but please feel free to reopen if the issue still occurs when testing with the the 3.6.0 beta 1 version that'll be available soon.

**Comment by frederoni** [comment]
The implementation above should yield a compile-time error saying “Control reaches end of non-void function”. Looks like warnings or errors about return type has been disabled in the build settings but it should be treated as an error by default.
`GCC_WARN_ABOUT_RETURN_TYPE = YES;`
![screenshot 2017-05-22 10 17 15](https://cloud.githubusercontent.com/assets/764476/26317809/06b8c288-3f19-11e7-84f7-df8792fd2a54.png)
